src/k2/cmd/tpcc/transactions.h

```diff
@@ -181,8 +202,78 @@ class PaymentT : public TPCCTxn
         return writeRow<History>(history, _txn).discard_result();
     }
 
+    // get customer ID by last name
+    future<int32_t> getCIdByLastName() {
+        k2::String lastName = _random.RandowLastNameString();
+        
+        return _client.createQuery(tpccCollectionName, "customer")
+        .then([this, &lastName](auto&& response) mutable {
+            K2EXPECT(response.status, dto::K23SIStatus::OK);
```

Comment:
You should use CHECK_READ_RESPONSE or create a new CHECK_QUERY_RESPONSE here so that the transaction can fail without exiting the TPC-C client. Same for the other uses of K2EXPECT.